grype-server/pkg/scanner/scanner.go

```diff
@@ -194,13 +208,20 @@ func (s *Scanner) loadBb(cfg *db.Config) error {
 	if status.Err != nil {
 		return fmt.Errorf("loaded DB has a failed status: %v", status.Err)
 	}
-	store, err := dbCurator.GetStore()
+	storeReader, dbCloser, err := dbCurator.GetStore()
+	if dbCloser != nil {
+		defer dbCloser.Close()
```

Comment:
are we sure we want it? will it not delete the DB from filesystem?
not sure what it does
```
func (s *store) Close() {
	s.db.Exec("VACUUM;")

	sqlDB, err := s.db.DB()
	if err != nil {
		_ = sqlDB.Close()
	}
}
```

https://github.com/anchore/grype/blob/d5b825e40bbfc4696e7220c94c9f4469466e305e/grype/db/v4/store/store.go#L265

Reply:
You are right it isn't necessary.
It closes the connections and frees the resources: https://github.com/anchore/grype/blob/f094b860b9b9ebcdf06d9b9c6f7d5c000fad395b/grype/db/db_closer.go#L5-L6
According to https://github.com/golang/go/blob/d11c58eedbee29b4912dd50508b36ad15ebb739e/src/database/sql/sql.go#L870-L876 it doesn't delete filesystem.